src/api/hooks/Membership.tsx

```diff
@@ -1,8 +1,39 @@
-import { useQuery } from 'react-query';
+import { useMutation, UseMutationResult, useQuery, useQueryClient } from 'react-query';
 import API from 'api/api';
-import { Membership } from 'types/Types';
+import { Membership, RequestResponse, User } from 'types/Types';
+import { MembershipType } from 'types/Enums';
 
 const QUERY_KEY = 'membership';
 export const useMemberships = (slug: string) => {
   return useQuery<Membership[]>([QUERY_KEY, slug], () => API.getMemberships(slug));
 };
+
+export const useCreateMembership = (): UseMutationResult<Membership, RequestResponse, { slug: string; user: User }, unknown> => {
+  const queryClient = useQueryClient();
+  return useMutation(({ slug, user }) => API.createMembership(slug, user.user_id), {
```

Comment:
```suggestion
export const useCreateMembership = (): UseMutationResult<Membership, RequestResponse, { slug: string; userId: string }, unknown> => {
  const queryClient = useQueryClient();
  return useMutation(({ slug, userId }) => API.createMembership(slug, userId), {
```

Noe grunn til at man må sende inn hele User hvis du kun trenger userId? Ville muligens også endret `slug` til `groupSlug` for å lettere forstå at det er slug til gruppe som er nødvendig